src/vulkan/pipelines.cpp

```diff
@@ -29,7 +31,8 @@ namespace IC::Renderer
 
         if (!file.is_open())
         {
-            throw std::runtime_error("failed to open file: " + filePath);
+            IC_CORE_ERROR("Failed to open file {0}.", filePath);
+            throw std::runtime_error("Failed to open file " + filePath + ".");
             return VK_NULL_HANDLE;
```

Comment:
I know this was already here, but wanted to point out that throwing and returning is redundant and the return will never be hit.